### PR TITLE
Migrate to Py3

### PR DESCRIPTION
**Pull request by alexjurkiewicz**

### CODE REVIEW

File: loaddb.py [hunk 1]
```diff
@@ -181,6 +176,7 @@ def run(self, cursor, elapsed):
 
 # These classes merely read lines from the logfile, and do not parse them.
 
+@functools.total_ordering # Implement __{gt,le,ge}__ based on __lt__ and __eq__
```
Comment (alexjurkiewicz) [inline loaddb.py:179]: Comparison code for this class was changed

File: loaddb.py [hunk 2]
```diff
@@ -1224,22 +1217,22 @@ def run_timers(c, elapsed_time):
     timer.run(c, elapsed_time)
 
 def load_extensions():
```
Comment (alexjurkiewicz) [inline loaddb.py:1219]: `imp` is deprecated, I hardcoded what it used to dynamically

File: query.py [hunk 3]
```diff
@@ -385,7 +385,7 @@ def get_death_causes(c):
       clean_rows.append( [ killer, r[2], r[3] ] )
     last = r
 
-  clean_rows.sort(lambda a,b: int(b[1] - a[1]))
+  clean_rows.sort(key=lambda i:i[1])
```
Comment (alexjurkiewicz) [inline query.py:388]: rewritten comparison

File: update_page.py [hunk 4]
```diff
@@ -45,11 +44,14 @@ def render(c, page, dest=None, pars=None, top_level_pars=False):
       template_data = t.render( attributes = pars )
   except:
     print("TEMPLATE ERROR")
-    print(mako.exceptions.text_error_template().render())
+    print((mako.exceptions.text_error_template().render()))
     raise
   f = open(target, 'w')
   try:
-    f.write(template_data)
+    if isinstance(template_data, bytes):
```
Comment (alexjurkiewicz) [inline update_page.py:51]: added defensive logic here (at least one caller was sending bytes)

File: htmlgen.py [hunk 5]
```diff
@@ -245,7 +245,7 @@ def pretty_dur(dur):
   try:
     secs = dur % 60
   except:
-    print("FAIL on %s" % dur)
```
Comment (alexjurkiewicz) [inline htmlgen.py:248]: I renamed this file to prevent issues with importing `mako.html` which popped up

File: loaddb.py [hunk 6]
```diff
@@ -194,16 +190,13 @@ def __init__(self, owner, filename, offset, time, xdict, processor):
...
-    # Descending time sort order, so that later dates go first.
-    if ltime > rtime:
-      return -1
-    elif ltime < rtime:
-      return 1
-    else:
-      return 0
+  def __lt__(self, other):
+    """Sorting only takes into account the time of a line."""
+    return self.time < other.time
```
Comment (ebering) [inline loaddb.py:195]: I think you got this backwards, based on the old comment. `cmp(x,y)` is negative when `x < y` and in this case that is when `self.time > other.time`

File: query.py (hunk 3 above)
Comment (ebering) [reply query.py:388]: Again I think you've reversed this one, `a < b` when `b[1] - a[1] < 0`, so this is a reverse sort on `i[1]`

File: loaddb.py (hunk 2 above)
Comment (ebering) [reply loaddb.py:1219]: Looks good

File: update_page.py (hunk 4 above)
Comment (ebering) [reply update_page.py:51]: Looks good

File: htmlgen.py (hunk 5 above)
Comment (ebering) [reply htmlgen.py:248]: Looks good

File: loaddb.py (hunk 6 above)
Comment (ebering) [reply loaddb.py:195]: Oops this should have been a reply but whatever.

**Comment by alexjurkiewicz** [comment]
Leaving this as a PoC. Let's come back after tournament.

**Pull request closed without merging**